solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,17 +19,37 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
+
     }
 
     @Override
     public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + filename);
+               ObjectOutputStream out = new ObjectOutputStream(fos)) {
+           out.writeObject(gameState);
+        } catch (FileNotFoundException ex) {
+            throw new SaveGameException("File not found", ex, SaveGameException.Type.USER, gameState);
+        } catch (IOException ex) {
+            throw new SaveGameException("I/O operation has been failed", ex, SaveGameException.Type.IO, gameState);
+        }
     }
 
     @Override
     public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+
+        MapState mapState = null;
+
+        try (FileInputStream fis = new FileInputStream(filesDirectory + filename);
+             ObjectInputStream inputStream = new ObjectInputStream(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a612985578>*